src/wandbot/rag/query_handler.py

```
@@ -241,75 +259,115 @@ def parse_output(
+            result, llm_api_status = await self._try_enhance_query(self.model, query, chat_history)
+            parsed_result = result.parse_output(query, chat_history)
+        except Exception as e:
+            logger.warning(f"Primary Query Enhancer model failed, trying fallback: {str(e)}")
+            try:
+                # Try fallback model
+                result, llm_api_status = await self._try_enhance_query(self.fallback_model, query, chat_history)
+                parsed_result = result.parse_output(query, chat_history)
+            except Exception as e:
+                logger.error(f"Both primary and fallback Query Enhancer models failed: {str(e)}")
+                # If both models fail, raise the error
+                raise Exception(f"Query enhancement failed: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)